src/backend/joanie/payment/models.py

```diff
@@ -31,13 +31,42 @@
 User = get_user_model()
 
 
-class ProformaInvoice(models.Model):
+class BaseModel(models.Model):
+    """Base model for all the models of the app payment"""
+
+    id = models.UUIDField(
+        verbose_name=_("id"),
+        help_text=_("primary key for the record as UUID"),
+        primary_key=True,
+        default=uuid.uuid4,
+        editable=False,
+    )
+    created_on = models.DateTimeField(
+        verbose_name=_("created on"),
+        help_text=_("date and time at which a record was created"),
+        auto_now_add=True,
+        editable=False,
+    )
+    updated_on = models.DateTimeField(
+        verbose_name=_("updated on"),
+        help_text=_("date and time at which a record was last updated"),
+        auto_now=True,
+        editable=False,
+    )
+
+    def save(self, *args, **kwargs):
+        """Enforce validation each time an instance is saved."""
+        self.full_clean()
+        super().save(*args, **kwargs)
```

Comment:
Why don't you use `joanie.core.models.BaseModel` ?

Reply:
because it's not in the same app, so i didn't think it was appropriate

Reply:
Don't know what is the best practice about this point.

But from my point of view, payment can not live without core app and BaseModels of those apps are equals so I'll say Don't Repeat Yourself.